tests/TestUtilities.py

```diff
@@ -6,22 +6,33 @@
 from coala_utils.ContextManagers import retrieve_stdout, retrieve_stderr
 
 
-def execute_coala(func, binary, *args):
+def execute_coala(func, binary, *args, debug=False):
     """
     Executes the main function with the given argument string from given module.
 
     :param function:    A main function from coala_json, coala_ci module etc.
     :param binary:      A binary to execute coala test
+    :param debug:       Run main function with ``debug=True`` and re-raise any
+                        exception coming back.
     :return:            A tuple holding a return value as first element,
                         a stdout output as second element and a stderr output
                         as third element if stdout_only is False.
     """
     sys.argv = [binary] + list(args)
     with retrieve_stdout() as stdout:
         with retrieve_stderr() as stderr:
-            retval = func()
+            try:
+                retval = func(debug=debug)
+            except Exception:
+                if debug:
+                    exc = sys.exc_info()
+                raise
```

Comment:
I wonder too... :D The problem are the `with` contexts... But I will think about simplification